core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -1764,6 +1773,7 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
     // Unset YARN mode system env variable, to allow switching between cluster types.
     System.clearProperty("SPARK_YARN_MODE")
     SparkContext.clearActiveContext()
+    _stopHooks.foreach(_())
```

Comment:
I'm not a big fan of adding more hooks, but since SQLContext doesn't have a `stop` method... anyway, probably good to wrap calls to the hooks with `Utils.tryLogNonFatalError`.